lib/adyen/hpp/signature.rb

```diff
@@ -13,7 +13,10 @@ module Signature
       #   params hash with the `sharedSecret` key.
       # @return [Hash] params The params that were passed in plus a new `merchantSig` param
       def sign(params, shared_secret = nil)
-        shared_secret ||= params.delete['sharedSecret']
+        params = params.dup
+        param_shared_secret = params.delete('sharedSecret')
+        shared_secret ||= param_shared_secret
+        params.delete('merchantSig')
         raise ArgumentError, "Cannot verify a signature without a shared secret" unless shared_secret
```

Comment:
This one is my error - the exception message should probably say something like `Cannot sign parameters without a shared secret` instead